userspace/libsinsp/sinsp_filtercheck_k8s.cpp

```diff
@@ -141,16 +141,16 @@ int32_t sinsp_filter_check_k8s::extract_arg(string_view fldname, string_view val
 {
 	int32_t parsed_len = 0;
 
-	if(val[fldname.size()] == '.')
+	if(val.size() > fldname.size() && val.at(fldname.size()) == '.')
 	{
 		size_t endpos;
 		for(endpos = fldname.size() + 1; endpos < val.length(); ++endpos)
 		{
-			if(!isalnum(val[endpos])
-				&& val[endpos] != '/'
-				&& val[endpos] != '_'
-				&& val[endpos] != '-'
-				&& val[endpos] != '.')
+			if(!isalnum(val.at(endpos))
+				&& val.at(endpos) != '/'
+				&& val.at(endpos) != '_'
+				&& val.at(endpos) != '-'
+				&& val.at(endpos) != '.')
```

Comment:
Why favoring `at()` over `operator[]`? The first performs bound checking, but in this case we are iterating inside the string `val`, so the bound checking is an extra calculation we could avoid.
Actually, all uses of `at()` in the PR seem to be after a manual bound check, so they could all be reverted to index operation without harm.